core/modules/editor/operations/text/wrap-selection.js

```diff
@@ -13,37 +13,121 @@ Text editor operation to wrap the selection with the specified prefix and suffix
 "use strict";
 
 exports["wrap-selection"] = function(event,operation) {
-	if(operation.selStart === operation.selEnd) {
-		// No selection; check if we're within the prefix/suffix
-		if(operation.text.substring(operation.selStart - event.paramObject.prefix.length,operation.selStart + event.paramObject.suffix.length) === event.paramObject.prefix + event.paramObject.suffix) {
+	var o = operation,
+		prefix = event.paramObject.prefix,
+		prefixLength = prefix.length,
+		suffix = event.paramObject.suffix,
+		suffixLength = suffix.length,
+		trimSelection = event.paramObject.trimSelection || "no",
+		selLength = o.selEnd - o.selStart;
+
+	var trailingSpaceAt = function(sel) {
+		// returns "yes", "start", "end", "no"
+		var _start,
+			_end,
+			result;
+		// this evaluation takes the user configuration into account!
+		switch (trimSelection) {
+			case "end":
+				result = (sel.trimEnd().length !== selLength) ? "end" : "no";
+			break;
+			case "yes":
+				_start = sel.trimStart().length !== selLength;
+				_end = sel.trimEnd().length !== selLength;
+				result = (_start && _end) ? "yes" : (_start) ? "start" : (_end) ? "end" : "no";
+			break;
+			case "start":
+				result = (sel.trimStart().length !== selLength) ? "start" : "no";
+			break;
+			default:
+				result = "no";
+			break;
+		}
+		return result;
+	}
+
+	function togglePrefixSuffix() {
+		// this was the only behavour till TW v5.3.3
+		if(o.text.substring(o.selStart - prefixLength, o.selStart + suffixLength) === prefix + suffix) {
 			// Remove the prefix and suffix
-			operation.cutStart = operation.selStart - event.paramObject.prefix.length;
-			operation.cutEnd = operation.selEnd + event.paramObject.suffix.length;
-			operation.replacement = "";
-			operation.newSelStart = operation.cutStart;
-			operation.newSelEnd = operation.newSelStart;
+			o.cutStart = o.selStart - prefixLength;
+			o.cutEnd = o.selEnd + suffixLength;
+			o.replacement = "";
+			o.newSelStart = o.cutStart;
+			o.newSelEnd = o.newSelStart;
 		} else {
 			// Wrap the cursor instead
-			operation.cutStart = operation.selStart;
-			operation.cutEnd = operation.selEnd;
-			operation.replacement = event.paramObject.prefix + event.paramObject.suffix;
-			operation.newSelStart = operation.selStart + event.paramObject.prefix.length;
-			operation.newSelEnd = operation.newSelStart;
+			o.cutStart = o.selStart;
+			o.cutEnd = o.selEnd;
+			o.replacement = prefix + suffix;
+			o.newSelStart = o.selStart + prefixLength;
+			o.newSelEnd = o.newSelStart;
 		}
-	} else if(operation.text.substring(operation.selStart,operation.selStart + event.paramObject.prefix.length) === event.paramObject.prefix && operation.text.substring(operation.selEnd - event.paramObject.suffix.length,operation.selEnd) === event.paramObject.suffix) {
+	}
+
+	// options: prefixLen, suffixLen
+	function removePrefixSuffix(options) {
+		options = options || {};
+		var prefixLen = options.prefixLen || 0;
```

Comment:
Using "Len" as a suffix is inconsistent with the code above. In a situation like this I'd prefer "lenPrefix".